pennylane/_qubit_device.py

```diff
@@ -370,7 +370,12 @@ def execute(self, circuit: QuantumTape, **kwargs):
         self.check_validity(circuit.operations, circuit.observables)
 
         # apply all circuit operations
-        self.apply(circuit.operations, rotations=circuit.diagonalizing_gates, **kwargs)
+        rotations = (
+            []
+            if self.capabilities().get("supports_non_diagonal_measurement", False)
+            else circuit.diagonalizing_gates
+        )
+        self.apply(circuit.operations, rotations=rotations, **kwargs)
```

Comment:
So I think this may be a bit overkill. We basically only don't want the diagonalizing gates if they are going to be expensive (aka a Sum class with overlapping wires). Instead here, we could manually get the diagonalizing gates from measurement processes if the observable isn't a `Sum` class. That way we would be maintaining the same behaviour in all cases except for the one we care about.

Reply:
chatted about this offline. removed from this PR, will address at a later date. default.qubit has its own funny way of dealing with diagonalizing gates (not) existing, and we should consider the best way to handle them, also considering DefaultQubit2 and any other device.